### PR TITLE
fix: arbitrum bridge withdraw status copies and button

### LINKED ISSUE #7894: Arbitrum Bridge withdraws displayed as success
### Overview

Currently in all envs, Arbitrum *withdraws* (ARB -> ETH) are detected as success as soon as the inbound Tx is complete:

image

However these take ~6-7 days to complete, and eventually require a claim.

That's as opposed to Arbitrum deposits (ETH -> ARB) which take ~5-15mn, and we have proper polling for success, meaning when we show success, the bridge *is* actually a success.

In an effort to make the experience feel more streamlined and less broken in the app, we should improve this from a product perspective so that this success step looks different from the regular one, and users are clearly informed that their swap is *not* complete but only their inbound is, as well as getting a gentle reminder (they have already been warned with an ack before bridging) that they will require a claim Tx in ~x days.

### References and additional details

Spotted while testing https://github.com/shapeshift/web/pull/7876

### Acceptance Criteria

- Success step for Arb withdraws doesn't look broken

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description

Updated copies inside `TradeSuccess`, made it agnostic so we could change it if we have another specific bridge in the future

Also added a button prop to `StepperStep` so we can pass a button as a prop that will be shown on the right

## Issue (if applicable)

closes #7894

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing
- Verify a regular swap that is not an arbitrum swap (verify the TradeSuccess screen copies and buttons)
- Do an arbitrum withdraw, check copies and click on the claim button to see if you are redirected to the claim view
- Ensure Arbitrum deposits (ETH -> ARB direction) do not display the claim CTA nor the Arbitrum withdraw specific copies

### Engineering
n/a

### Operations
n/a

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)
image
image

### Regular
image
image
image

### CODE REVIEW

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx [hunk 1]
```diff
@@ -128,7 +133,15 @@ export const MultiHopTradeConfirm = memo(() => {
...
         {isTradeComplete ? (
-          <TradeSuccess handleBack={handleBack}>
+          <TradeSuccess
+            handleBack={handleBack}
+            titleTranslation={
+              isArbitrumBridgeWithdraw ? 'bridge.arbitrum.success.tradeSuccess' : undefined
+            }
+            descriptionTranslation={
+              isArbitrumBridgeWithdraw ? 'bridge.arbitrum.success.tradeComplete' : undefined
+            }
```
Comment (NeOMakinG) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx:143]: I wanted to put this inside some memos but it makes no points for now as this is the only bridge we need this system for at this point

File: src/assets/translations/en/main.json [hunk 2]
```diff
@@ -793,7 +793,11 @@
       "description": "Bridge via Arbitrum",
       "waitWarning": "Arbitrum’s native bridge is more secure but can take up to a week to claim.",
       "waitCta": "I understand that it will take ~7 days before I can claim funds on Ethereum",
-      "claimCta": "I understand that after claiming my funds, I’ll have to send another transaction with a fee"
+      "claimCta": "I understand that after claiming my funds, I’ll have to send another transaction with a fee",
+      "success": {
+        "tradeSuccess": "Bridge initiated",
+        "tradeComplete": "Successfully initiated bridging %{symbol} to %{chainName}."
```
Comment (gomesalexandre) [inline src/assets/translations/en/main.json:799]: suggestion: can we add withdraw terminology here? Deposit is an actual success, not a "bridge initiated"

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 3]
```diff
@@ -281,6 +281,7 @@ export const TradeInput = ({ isCompact, tradeInputRef, onChangeTab }: TradeInput
     },
     [dispatch],
   )
+  console.log({ buyAsset })
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:284]: ser

File: src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx [hunk 4]
```diff
@@ -41,11 +51,17 @@ export const TradeSuccess = ({ handleBack, children }: TradeSuccessProps) => {
 
     const chainName = adapter.getDisplayName()
 
+    if (descriptionTranslation)
+      return translate(descriptionTranslation, {
+        symbol: lastHop.buyAsset.symbol,
+        chainName,
+      })
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx:58]: preferably-blocking: this isn't generalized and is tying the translation interpolation options to `descriptionTranslation`. Can we either:
- pass `translation` from up top as a string
- pass translation args down as `titleTranslation` and `descriptionTranslation`

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx:143]: `titleTranslation` and `descriptionTranslation` is misleading here, this is a translation key without interpolation args. See https://github.com/shapeshift/web/pull/7992/files#r1813308736

File: src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx [hunk 5]
```diff
@@ -18,11 +18,21 @@ import { useAppSelector } from 'state/store'
 
 import { TwirlyToggle } from '../TwirlyToggle'
 
-export type TradeSuccessProps = { handleBack: () => void; children: JSX.Element }
+export type TradeSuccessProps = {
+  handleBack: () => void
+  children: JSX.Element
+  titleTranslation?: string
+  descriptionTranslation?: string
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx:25]:
```suggestion
  titleTranslation?: string | [string, InterpolationOptions]
  descriptionTranslation?: string | [string, InterpolationOptions]
```

**Review by gomesalexandre — APPROVED** [review]
Some general cleanup comments to tackle at your convenience, but LGTM functionally:

- Arb Withdraws copy/button are happy

https://jam.dev/c/e03cb590-fe62-4f3f-9568-06e815459d2b

@NeOMakinG noticed one oddity here re: claim not being reflected before refresh, though Tx wasn't picked by ws either which could explain why, non-blocking and can tackle in a follow-up if repro'd on prod

- Arb deposits are happy and showing generic copy and no button

Unable to test because of lack of funds, deferring to ops here

- Other trades display the generic copy and no button

https://jam.dev/c/41ca54d8-76ea-44c6-bf72-5f2300496739

File: src/assets/translations/en/main.json (hunk 2 above)
Comment (NeOMakinG) [reply src/assets/translations/en/main.json]: Done in [3d19513](https://github.com/shapeshift/web/pull/7992/commits/3d1951386bf5e0dba41a1d9082f2a0598502f0ae)

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx (hunk 3 above)
Comment (NeOMakinG) [reply src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx]: Sry in [3d19513](https://github.com/shapeshift/web/pull/7992/commits/3d1951386bf5e0dba41a1d9082f2a0598502f0ae)

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx (hunk 1 above)
Comment (NeOMakinG) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx]: Done in [3d19513](https://github.com/shapeshift/web/pull/7992/commits/3d1951386bf5e0dba41a1d9082f2a0598502f0ae) I guess if I understood it right

File: src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx (hunk 4 above)
Comment (NeOMakinG) [reply src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx]: Done in [3d19513](https://github.com/shapeshift/web/pull/7992/commits/3d1951386bf5e0dba41a1d9082f2a0598502f0ae)

File: src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx (hunk 5 above)
Comment (NeOMakinG) [reply src/components/MultiHopTrade/components/TradeSuccess/TradeSuccess.tsx]: Done in [3d19513](https://github.com/shapeshift/web/pull/7992/commits/3d1951386bf5e0dba41a1d9082f2a0598502f0ae)

**Comment by NeOMakinG** [comment]
> Some general cleanup comments to tackle at your convenience, but LGTM functionally:
> 
> * Arb Withdraws copy/button are happy
> 
> https://jam.dev/c/e03cb590-fe62-4f3f-9568-06e815459d2b
> 
> @NeOMakinG noticed one oddity here re: claim not being reflected before refresh, though Tx wasn't picked by ws either which could explain why, non-blocking and can tackle in a follow-up if repro'd on prod
> 
> * Arb deposits are happy and showing generic copy and no button
> 
> Unable to test because of lack of funds, deferring to ops here
> 
> * Other trades display the generic copy and no button
> 
> https://jam.dev/c/41ca54d8-76ea-44c6-bf72-5f2300496739

Happy to fix this in a follow up when we deliver it and reproduce in prod

**Pull request merged**